### PR TITLE
Do not allow login unless user has namespace permissions

### PR DESCRIPTION
**Pull request by abought**
Refs https://openscience.atlassian.net/browse/LEI-117
Companion to https://github.com/CenterForOpenScience/experimenter/pull/69

See companion PR for description.

### CODE REVIEW

File: exp-models/addon/authenticators/jam-osf-jwt.js [hunk 1]
```diff
@@ -34,6 +48,14 @@ export default Base.extend({
         return this._post(accessToken).then(function(res) {
             res.data.attributes.accessToken = accessToken;
             return res.data.attributes;
+        })
+            .then((res) => {
```
Comment (chrisseto) [inline exp-models/addon/authenticators/jam-osf-jwt.js]: Remove the extra new line here.
`}).then(res => ...`

File: exp-models/addon/authenticators/jam-osf-jwt.js [hunk 2]
```diff
@@ -34,6 +48,14 @@ export default Base.extend({
...
+               // Only allow someone to login to the admin panel if they have have namespace-level permissions for this project
+                return this._askNamespace(res.token)
+                    .then(() => {return res;})
```
Comment (chrisseto) [inline exp-models/addon/authenticators/jam-osf-jwt.js]: You don't need the `{}` or the `return` here.
`.then(() => res)` will do the same thing

File: exp-models/addon/authenticators/jam-osf-jwt.js [hunk 3]
```diff
@@ -34,6 +48,14 @@ export default Base.extend({
...
+                    .fail(()=> {
```
Comment (chrisseto) [inline exp-models/addon/authenticators/jam-osf-jwt.js]: See above.

Please add a space between `()` and `=>`

**Comment by abought** [comment]
While I would have slightly preferred to not open the laptop tonight, careful reading deserves a timely response. :) All 3 requested changes pushed.

**Pull request merged**